CRT.c

```diff
@@ -957,25 +960,39 @@ int CRT_scrollWheelVAmount = 10;
 
 ColorScheme CRT_colorScheme = COLORSCHEME_DEFAULT;
 
-ATTR_NORETURN
+
 static void CRT_handleSIGTERM(int sgn) {
+   CRT_terminateSignal = (sig_atomic_t)sgn;
+   CRT_terminateRequested = 1;
+}
+
+static void CRT_tryTermExit(void) {
+
+   if (!CRT_terminateRequested) {
+      return;
+}
+   CRT_terminateRequested = 0;
```

Comment:
What if a signal happens between this line and the `if (!CRT_terminateRequested)` conditional?